src/pages/organize/[orgId]/people/incoming/index.tsx

```diff
@@ -97,25 +48,22 @@ const IncomingPage: PageWithLayout<Props> = ({ orgId }) => {
         justifyContent="flex-end"
         sx={{ mr: 2, my: 2 }}
       >
-        {/* filter by form name */}
-        <FormControl sx={{ minWidth: 200 }}>
-          <InputLabel>{messages.forms()}</InputLabel>
-          <Select
-            label={messages.forms()}
-            onChange={(event) => setFilterByForm(event.target.value)}
-            placeholder={messages.forms()}
-            value={filterByForm}
-          >
-            <MenuItem selected value={'all'}>
-              {messages.submissionPane.allForms()}
-            </MenuItem>
-            {uniqueFormTitles.map((title, index) => (
-              <MenuItem key={index} value={title}>
-                {title}
-              </MenuItem>
-            ))}
-          </Select>
-        </FormControl>
+        <ZUIFuture
+          future={joinSubmissions}
+          ignoreDataWhileLoading
+          skeleton={<JoinFormSelect />}
+        >
+          {(submissions) => (
+            <JoinFormSelect
+              formId={filterByForm}
+              forms={uniqBy(
+                submissions.map((s) => s.form),
+                'id'
+              )}
```

Comment:
I first mistook this as creating a unique list of submissions (and didn't realize the argument was first mapped to a list of forms). You don't have to change this, but in cases like this I find it helpful to have a variable called something like `uniqueForms` declared above with the return value from this exact expression, and then just use the variable down here. The name of the variable then helps me (as a reader of the code) to understand what the expression does.

Reply:
✨ https://github.com/zetkin/app.zetkin.org/pull/2090/commits/206197f91b24f1a4cb4e7ad8b29bc54adc2e1d0b ✨